### PR TITLE
[IMP] charts: disable legend for single dataset

### PR DESCRIPTION
**Pull request by anhe-odoo**
## Task Description

When drawing a chart with only one dataset, we are allowed to choose the position of the legend but no legend will appears even after changing the position. This commit aims to fix thie issue, disabling the select box allowing to change the legend position.

The chart padding has also been adapted to have more padding when there is no title/legend.

## Related Task

- Task: [4028903](https://www.odoo.com/odoo/project/2328/tasks/4028903?cids=1)

## review checklist

- [ ] feature is organized in plugin, or UI components
- [ ] support of duplicate sheet (deep copy)
- [ ] in model/core: ranges are Range object, and can be adapted (adaptRanges)
- [ ] in model/UI: ranges are strings (to show the user)
- [ ] undo-able commands (uses this.history.update)
- [ ] multiuser-able commands (has inverse commands and transformations where needed)
- [ ] new/updated/removed commands are documented
- [ ] exportable in excel
- [ ] translations (\_t("qmsdf %s", abc))
- [ ] unit tested
- [ ] clean commented code
- [ ] track breaking changes
- [ ] doc is rebuild (npm run doc)
- [ ] status is correct in Odoo

### CODE REVIEW

File: src/helpers/figures/charts/bar_chart.ts [hunk 1]
```diff
@@ -238,8 +238,14 @@ function getBarConfiguration(
     legend.position = chart.legendPosition;
   }
   config.options.plugins!.legend = { ...config.options.plugins?.legend, ...legend };
+  let topPadding = 25;
```
Comment (hokolomopo) [inline src/helpers/figures/charts/bar_chart.ts:241]: Maybe create a helper, copy/pasting 3 times seems a bit much :)

File: src/components/side_panel/chart/chart_with_axis/design_panel.xml [hunk 2]
```diff
@@ -9,7 +9,7 @@
           <t t-set-slot="title">Legend position</t>
           <select
             t-att-value="props.definition.legendPosition ?? 'top'"
-            class="o-input"
```
Comment (hokolomopo) [inline src/components/side_panel/chart/chart_with_axis/design_panel.xml:12]: The class `legend-position-selector` doesn't seems to be used anywhere I think ?

File: src/helpers/figures/charts/chart_common.ts [hunk 3]
```diff
@@ -428,3 +428,21 @@ export function getDefinedAxis(definition: ChartWithAxisDefinition): {
   useLeftAxis ||= !useRightAxis;
   return { useLeftAxis, useRightAxis };
 }
+
+export function computeChartPadding(
+  displayTitle: boolean,
+  displayLegend: boolean
+): {
```
Comment (pro-odoo) [inline src/helpers/figures/charts/chart_common.ts:435]:
```suggestion
export function computeChartPadding({
  displayTitle: boolean,
  displayLegend: boolean
}): {
```
IMO it's easier to read a call like 
```ts
padding: computeChartPadding({
    displayTitle: !!chart.title.text,
    displayLegend: legend.display !== false
}),
``` 
than 
```ts
padding: computeChartPadding(!!chart.title.text, legend.display !== false),
```

**Review by pro-odoo — CHANGES_REQUESTED** [review]
Just a little detail

**Review by hokolomopo — COMMENTED** [review]
Maybe also add some tests on the chart padding, and It'll be good :)

**Review by LucasLefevre — APPROVED** [review]
robodoo r+

**Pull request closed without merging**